2004-11-30  Jeff Johnston  <jjohnstn@redhat.com> 

	* breakpoint.c (remove_breakpoints): Continue removing breakpoints
	even if an error occurs.  Remove a failure code for the last failure
	only.

--- gdb+dejagnu-20040607/gdb/breakpoint.c.fix2	Tue Nov 30 18:01:33 2004
+++ gdb+dejagnu-20040607/gdb/breakpoint.c	Tue Nov 30 18:06:01 2004
@@ -1297,6 +1297,7 @@ remove_breakpoints (void)
 {
   struct bp_location *b;
   int val;
+  int return_val = 0;
 
   ALL_BP_LOCATIONS (b)
   {
@@ -1304,10 +1305,10 @@ remove_breakpoints (void)
       {
 	val = remove_breakpoint (b, mark_uninserted);
 	if (val != 0)
-	  return val;
+	  return_val = val;
       }
   }
-  return 0;
+  return return_val;
 }
 
 int
